book/forms.rst

```diff
@@ -1568,6 +1568,15 @@ are 4 possible *parts* of a form that can be rendered:
 
 .. note::
 
+    The portion corresponding to ``collection`` type is ``entry``.
```

Comment:
@Th3Mouk I'm sorry I thought your example was for the doc and not about illustrating what you think is missing in the docs :)

In fact, I think there is a misunderstanding about the ``entry`` portion in your phrasing.

The portion corresponding to `collection` is `collection` (see [here](https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig#L26)) which can get overridden by default with the field name, in your example `periodeSemaine`:
```php
$builder->add('periodeSemaines', CollectionType::class, // ...
```

But you can define it using the `block_name` option (ref [doc](http://symfony.com/doc/current/reference/forms/types/form.html#block-name)).

`CollectionType` actually overrides the `block_name` value for its entries options with `entry` using a normalizer (see [here](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Extension/Core/Type/CollectionType.php#L85)).

So in your case (I presume):
```php
// Controller
$form = $this->createBuilder()
    ->add('calendrier_exercice', CalendrierExerciceType::class, array(
    // let you override 'form_row' with '_calendrier_exercice_row'
    // ...

// CalendrierExerciceType.php
$builder->add('periodeSemaines', CollectionType::class, array(
    // 'collection' => 'periodeSemaines'
    'entry_type' => PeriodeSemaineType::class,
    // 'form' => 'entry', overridden for all entries
    // ...

// PeriodeSemaineType
$builder->add('semaines', CollectionType::class, array(
    // 'collection' => '_semaines'
    'entry_type' => SemaineZoneType::class,
    // 'form' => 'entry'
    // ...
```
So you get:
 `form_collection_entry_collection_entry` default "cascade" block name which you can override with `_calendrier_exercice_periodeSemaines_entry_semaines_entry`

Note that you could have:
```php
// PeriodeSemaineType
$resolver->setDefault('block_name', 'periode_semaine');

// SemainZoneType
$resolver->setDefault('block_name', 'semaine');
```

and just do:
```twig
{# no underscore needed as 'block_name' is explicitly set #}
{% block periode_semaine_entry_row %}
{% block semaine_entry_row %}
```

Conclusion
========
`entry` will always override `block_name` option even defined in a form type if it's nested in a `CollectionType`.

And you're right that should be mentioned in the docs! ;)

Perhaps it should also be mentioned in the `CollectionType` reference as well. Could be a note near `entry_options` or in the introduction ?

Reply:
This is all quite complex. Can you summarize this down to a note we should add? Or is it easier to say: "If you want to override a part of a form, go to the form profiler, find the field, and look at the block prefixes section". We could even add some screenshots. This seems maybe simpler than explaining the logic (unless the logic is quite simple - I'm arriving here late)